### DISCUSSION TITLE
Locked and Alarm status not matching

### DISCUSSION THREAD

**Discussion by Arne1346** [ianjwhite99/sync-connect-sdk#31] [Q&A] [answered]
Hi Ian,

thank you for your great work!
Since a month I have a 2022 Ford Galaxy, downloaded your toolkit and started my journey.

I am surprised about the Lock-status as well as the alarm status. The Ford API always reports "unlocked" and "not armed" even if the car is locked properly either by button press on the key or touching the door handle.
Only after pressing the lock button in the Ford SycConnect app the API also reports back the car as "locked" and alarm "armed".

Did you made a similar experience and maybe have an idea?

Thank you,
Arne

**Comment by HKLM** [comment, marked as answer]
> I am surprised about the Lock-status as well as the alarm status. The Ford API always reports "unlocked" and "not armed" even if the car is locked properly either by button press on the key or touching the door handle. Only after pressing the lock button in the Ford SycConnect app the API also reports back the car as "locked" and alarm "armed".
> 
> Did you made a similar experience and maybe have an idea?

Thats normally how it goes. I get the same thing for my 2021 F-250.  What I think is happening, 

1. That when you turn off the vehicle. 
2. The vehicle sends it's current status AT THAT moment to the Ford API server
3. You exit the vehicle, 
4. Push the lock button on key/key fob to lock and arm the alarm
4. After x amount of pre-arming time, the alarm is armed
5. You query the Ford API for the vehicle status-> Ford API says `lockStatus=UNLOCKED` and  `alarm=NOTSET` 

This is because the vehicle has NOT sent any more data to Ford API server, since step 2!

FYI when you send the "status" command, you are "asking the Ford API for the status".  The Ford API server responds with the info, that is in the servers record (data from the last time the vehicle sent a update) This is not the same as "asking the vehicle" for the status, 

There is currently two ways, that I know of, to get the Ford API server to update the info it has on the vehicle (without unlocking or disarming the alarm). You can manually force the vehicle to tell the Ford API it's current status, by sending the "request refresh" command (by using the following code below) or send the "lock" command. Either one of those two will have the vehicle, send a new current status to Ford API. Now send a "status" command and the Ford API will respond  `lockStatus=LOCKED` and  `alarm=SET` 


Add this to the [vehicle.py](https://github.com/ianjwhite99/sync-connect-sdk/blob/stable/syncconnect/vehicle.py) file, 
inside the Vehicle class
add

       
````
    def request_update(self):
        """ PUT Vehicle.status

        Returns:
            Response: Send request to refresh data from the cars module.
            Note: Next need to call status() to receive the updated data
        Raises:
            SyncException

        """

        response = self.api.put(
            const.API_URL,
            'vehicles/v2/' +
            self.vehicle_id +
            '/status', None)
        return response.json()
````

**Comment by ianjwhite99** [comment]
> Hi Ian,
> 
> thank you for your great work! Since a month I have a 2022 Ford Galaxy, downloaded your toolkit and started my journey.
> 
> I am surprised about the Lock-status as well as the alarm status. The Ford API always reports "unlocked" and "not armed" even if the car is locked properly either by button press on the key or touching the door handle. Only after pressing the lock button in the Ford SycConnect app the API also reports back the car as "locked" and alarm "armed".
> 
> Did you made a similar experience and maybe have an idea?
> 
> Thank you, Arne

Hi @Arne1346,

Thank you for your kind words. I apologize for not getting back to you sooner, I didn't see the notification of the post in the discussion board. @HKLM is correct. The only possible way to get the correct status is by sending a refresh request to the vehicle. This allows the vehicle to re-sync with Fords backend. At that point you can make another status request to the vehicle and grab the updated information.

**Reply by ianjwhite99** [reply]
@HKLM Please re-fork the repo and re-open your Pull Request. I rebased the commit history to clean up the commits, since they were all over the place with an unverified account.

**Comment by ianjwhite99** [comment]
@Arne1346 The latest version has just been released. This updates the `refresh_status()` method with the changes that @HKLM proposed. You can now correctly request and retrieve the vehicles current status.
